optuna_dashboard/static/action.ts

```diff
@@ -33,8 +33,23 @@ export const actionCreator = () => {
   }
 
   const updateStudyDetail = (studyId: number) => {
-    getStudyDetailAPI(studyId)
+    let nLocalFixedTrials = 0
+    if (studyId in studyDetails) {
+      for (const trial of studyDetails[studyId].trials) {
+        if (!["Running", "Waiting"].includes(trial.state)) {
+          nLocalFixedTrials += 1
+        } else {
+          break
+        }
+      }
+    }
```

Comment:
You may be able to simplify this by using `array.findIndex()`.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex